analyzer/src/main/kotlin/managers/utils/NuGetInspector.kt

```diff
@@ -0,0 +1,324 @@
+    filter { it.role == "author" }.mapNotNullTo(sortedSetOf()) { party ->
+        buildString {
+            party.name?.let { append(it) }
+            party.email?.let {
+                append(if (party.name != null && it.isNotBlank()) " <$it>" else it)
+            }
+        }.takeIf { it.isNotBlank() }
+    }
+
+private fun NuGetInspector.PackageData.toPackageReference() =
+    PackageReference(
+        id = Identifier(type = TYPE, namespace = "", name = name, version = version),
+        dependencies = dependencies.mapTo(sortedSetOf()) {
+            PackageReference(
+                id = Identifier(type = TYPE, namespace = "", name = it.name, version = it.version),
+            )
+        }
+    )
+
+internal fun NuGetInspector.Result.toOrtProject(
+    managerName: String,
+    analysisRoot: File,
+    definitionFile: File,
+): Project {
+    val spec = resolveLocalSpec(definitionFile)?.let { NuGetSupport.XML_MAPPER.readValue<PackageSpec>(it) }
+    val id = Identifier(
+        type = managerName,
+        namespace = "",
+        name = spec?.metadata?.id ?: definitionFile.relativeTo(analysisRoot).invariantSeparatorsPath,
+        version = spec?.metadata?.version.orEmpty()
+    )
+    val workingDir = definitionFile.parentFile
+
+    val nestedPackages: MutableList<NuGetInspector.PackageData> = mutableListOf()
+
+    packages.forEach {
+        it.packages.forEach { pkg -> nestedPackages.add(pkg) }
+    }
+
+    val packageReferences = nestedPackages.mapTo(sortedSetOf()) {
+        it.toPackageReference()
+    }
+
+    val scopes = sortedSetOf(Scope(headers.first().projectFramework, packageReferences))
+    return Project(
+        id = id,
+        definitionFilePath = VersionControlSystem.getPathInfo(definitionFile).path,
+        vcs = VcsInfo.EMPTY,
+        authors = parseAuthors(spec),
+        vcsProcessed = PackageManager.processProjectVcs(workingDir),
+        declaredLicenses = parseLicenses(spec),
+        homepageUrl = "",
+        scopeDependencies = scopes
+    )
+}
+
+internal fun Set<NuGetInspector.PackageData>.toOrtPackages(): Set<Package> =
+    groupBy { "${it.name}:${it.version}" }.mapTo(mutableSetOf()) { (_, packages) ->
+        val pkg = packages.first()
+
+        fun NuGetInspector.PackageData.getHash(): Hash = Hash.create(
+            sha512?.lowercase() ?: sha256?.lowercase() ?: sha1?.lowercase() ?: md5?.lowercase() ?: ""
+        )
+
+        val id = Identifier(type = TYPE, namespace = pkg.namespace.orEmpty(), name = pkg.name, version = pkg.version)
+        val declaredLicenses: SortedSet<String> = sortedSetOf()
+        val pkgDeclaredLicense = pkg.declaredLicense.orEmpty()
+
+        if (pkgDeclaredLicense.isNotBlank()) {
+            val licenseData = yamlMapper.readValue<Map<String, String>>(pkgDeclaredLicense)
+            val type = licenseData["LicenseType"].orEmpty()
+            licenseData["LicenseExpression"]?.also { if (type == "Expression") declaredLicenses += it }
```

Comment:
"Expression" is a lowercase value
```suggestion
            licenseData["LicenseExpression"]?.also { if (type == "expression") declaredLicenses += it }
```

Reply:
Done!

Reply:
Thanks